helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixManager.java

```diff
@@ -131,6 +131,7 @@
   private int _reportLatency;
 
   protected RealmAwareZkClient _zkclient;
+  private RealmAwareZkClient.RealmAwareZkConnectionConfig _realmAwareZkConnectionConfig;
```

Comment:
As I pinged you in slack, if we can let RealmAwareZkConnectionConfig backward supports _zkAddress, then we can safe tons of validations and branches code in this class.
Moreover, we can simplify the code following this principle,
1. new APIs only accept RealmAwareZkConnectionConfig, we don't use zkAddress directly.
2. for older API, we create default RealmAwareZkConnectionConfig based on the zkAddress. So we don't have 2 possibilities in the internal code implementations.

Reply:
That goes against the design of "RealmAware"ZkClient/configs. They are not tied to a single zk, and therefore we should not have a ZK field.